### DISCUSSION TITLE
Using system Environment Variables

### DISCUSSION THREAD

**Discussion by JarneDeSmet** [vercel/community#6486] [Help] [answered]
### Summary

Hi, I am deploying a Next project to vercel using custom github actions. I need to access the VERCEL_GIT_COMMIT_SHA in my redux store. I am not sure why it is not working. i have tried using the NEXT_PUBLIC prefix. I tried accessing through the nextConfig. 

i either get an undefined or just an empty field

### Steps to Reproduce

I can not give access to my repo or code but setting up a next project with redux should be good enough. then try to console log the env variable and deploy to vercel.

**Comment by versecafe** [comment, marked as answer]
@JarneDeSmet these variables are set during the deployment process on Vercel so would only be accessible in your Vercel deployments, but if you're running through github actions you should be able to get the GIT_COMMIT_SHA from the action runner instead to use.

**Reply by yashsway** [reply]
@versecafe is this still true if I'm kicking of the build in CircleCI through `vercel build`? When attempting to debug the environment variables that Vercel provides, I get this in CircleCI logs through some `console.log`'s in the webpack config I'm using to make the build:

```
debugging {
*****************:build:stage:   GIT_COMMIT_SHA: undefined, // process.env.VERCEL_GIT_COMMIT_SHA
*****************:build:stage:   GIT_COMMIT_REF: undefined, // process.env.VERCEL_GIT_COMMIT_REf
*****************:build:stage:   GITHUB_PR_ID: undefined, // process.env.VERCEL_GIT_PULL_REQUEST_ID
*****************:build:stage:   VERCEL: '1', // process.env.VERCEL
*****************:build:stage:   VERCEL_ENV: 'preview', // process.env.VERCEL_ENV
*****************:build:stage:   DEPLOYMENT_ID: undefined // process.env.DEPLOYMENT_ID
}
```

The `1` for `process.env.VERCEL` tells me it's populating system environment variables correct? Why am I not getting them?

Running this build through a Turborepo command in the Vercel project settings.. so the top of the log looks like this for more context:
```
Vercel CLI 34.0.0
Warning: Detected "engines": { "node": ">=18.0.0" } in your `package.json` that will automatically upgrade when a new major Node.js Version is released. Learn More: http://vercel.link/node-version
Detected `pnpm-lock.yaml` version 9 generated by pnpm 8
Running "install" command: `[ ! -d node_modules ] && pnpm install --frozen-lockfile || echo pnpm-lock.yaml unchanged. Using cache.`...
pnpm-lock.yaml unchanged. Using cache.

> REDACTED@0.0.1 build:stage /home/circleci/project
> turbo run build:stage "--filter=*****************"

Attention:
Turborepo now collects completely anonymous telemetry regarding usage.
This information is used to shape the Turborepo roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://turbo.build/repo/docs/telemetry

• Packages in scope: *****************
• Running build:stage in 1 packages
• Remote caching enabled

*****************:build:stage: cache miss, executing 549c204e81166391
*****************:build:stage: 
*****************:build:stage: > *****************@2.296.0 build:stage /home/circleci/project/apps/*****************
*****************:build:stage: > env-cmd --file .env.staging webpack --config webpack.production.config.js --progress=profile --color
```
